ui/src/common/SettingsDialog.tsx

```diff
@@ -1,68 +1,64 @@
-import Button from '@material-ui/core/Button';
-import Dialog from '@material-ui/core/Dialog';
-import DialogActions from '@material-ui/core/DialogActions';
-import DialogContent from '@material-ui/core/DialogContent';
-import DialogTitle from '@material-ui/core/DialogTitle';
-import TextField from '@material-ui/core/TextField';
-import Tooltip from '@material-ui/core/Tooltip';
-import React, {Component} from 'react';
-import {observable} from 'mobx';
-import {observer} from 'mobx-react';
-import {inject, Stores} from '../inject';
+import React, {useState} from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import Tooltip from '@mui/material/Tooltip';
+import {useAppDispatch} from '../store';
+import {changePassword} from '../store/auth-actions.ts';
 
 interface IProps {
     fClose: VoidFunction;
 }
 
-@observer
-class SettingsDialog extends Component<IProps & Stores<'currentUser'>> {
-    @observable
-    private pass = '';
+// TODO: rename to ChangePasswordDialog
```

Comment:
Please make as few refactoring as possible in this PR, only what's necessary so it says small.

Reply:
ok, I removed that todo again and we keep the component as a function component